repair/repair.cc

```diff
@@ -1311,6 +1377,31 @@ static future<> repair_ranges(lw_shared_ptr<repair_info> ri) {
     });
 }
 
+static future<> try_wait_for_hints_to_be_replayed(repair_uniq_id id, std::vector<gms::inet_address> source_nodes, std::vector<gms::inet_address> target_nodes) {
+    auto& sp = service::get_local_storage_proxy();
+    rlogger.info("repair id {}: started replaying hints before repair, target nodes: {}", id, target_nodes);
+    try {
+        seastar::abort_source combined_as;
+        auto attach = [&] (seastar::abort_source& as) {
+            as.check();
+            return as.subscribe([&] () noexcept {
+                if (!combined_as.abort_requested()) {
+                    combined_as.request_abort();
+                }
+            });
+        };
+
+        const auto shutdown_sub = attach(repair_tracker().get_shutdown_abort_source());
+        const auto abort_all_sub = attach(repair_tracker().get_abort_all_abort_source());
+
+        co_await sp.wait_for_hints_to_be_replayed(id.uuid, std::move(source_nodes), std::move(target_nodes), combined_as);
+        rlogger.info("repair id {}: finished replaying hints, continuing with repair", id);
```

Comment:
It will be useful to log how long it takes to reply the hints. Since this time is now added to repair time.

Reply:
Done